server/manager/src/main/java/org/apache/accumulo/manager/tableOps/Utils.java

```diff
@@ -166,8 +168,17 @@ private static Lock getLock(ServerContext context, AbstractId<?> id, long tid,
     var fLockPath =
         FateLock.path(context.getZooKeeperRoot() + Constants.ZTABLE_LOCKS + "/" + id.canonical());
     FateLock qlock = new FateLock(context.getZooReaderWriter(), fLockPath);
-    Lock lock = DistributedReadWriteLock.recoverLock(qlock, lockData);
-    if (lock == null) {
+    DistributedLock lock = DistributedReadWriteLock.recoverLock(qlock, lockData);
+    if (lock != null) {
+
+      // Validate the recovered lock type
+      boolean isWriteLock = lock.getType() == LockType.WRITE;
```

Comment:
This change does not need to be made in this PR.  Looking at this code, wondering if it would clean things up to pass LockType into this method instead of  `boolean writeLock`. It would make the comparison in this if stmt nicer, but not sure if its an overall win.  If its seems useful, could be a follow on PR.

Reply:
I had the same thought, but it looks like it would be one of those changes that starts out small, and then starts leaking out into a lot more code. It does seem useful, but definitely as a separate change.

Reply:
`private static Lock getLock(ServerContext context, AbstractId<?> id, long tid,
      LockType lockType) {
    byte[] lockData = FastFormat.toZeroPaddedHex(tid);
    var fLockPath =
        FateLock.path(context.getZooKeeperRoot() + Constants.ZTABLE_LOCKS + "/" + id.canonical());
    FateLock qlock = new FateLock(context.getZooReaderWriter(), fLockPath);
    DistributedLock lock = DistributedReadWriteLock.recoverLock(qlock, lockData);
    if (lock != null) {

      // Validate the recovered lock type
      if (lock.getType() != lockType) {
        throw new IllegalStateException("Unexpected lock type " + lock.getType()
            + " recovered for transaction " + FateTxId.formatTid(tid) + " on object " + id
            + ". Expected " + lockType + " lock instead.");
      }
    } else {
      DistributedReadWriteLock locker = new DistributedReadWriteLock(qlock, lockData);
      if (lockType == LockType.WRITE) {
        lock = locker.writeLock();
      } else {
        lock = locker.readLock();
      }
    }
    return lock;
  }`

Maybe with this change, the method is more flexible, and you can pass the desired LockType directly instead of using a boolean parameter. Any suggestions?

Reply:
> Maybe with this change, the method is more flexible, and you can pass the desired LockType directly instead of using a boolean parameter. Any suggestions?

@rsingh433 something along those lines would be good, not sure how much it impacts the calling code.  Would be good to try the change and see what it looks like.